crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -45,8 +45,11 @@ impl<'ast> Visit<'ast> for AstScanner<'ast> {
   fn visit_identifier_reference(&mut self, ident: &IdentifierReference) {
     let symbol_id = self.resolve_symbol_from_reference(ident);
     match symbol_id {
-      Some(symbol_id) if self.is_top_level(symbol_id) => {
-        self.add_referenced_symbol(symbol_id);
+      Some(symbol_id) => {
+        self.try_diagnostic_forbid_const_assign(symbol_id);
```

Comment:
Since you just need the `Symbold`, you could call this method in `visit_binding_identifier`. Using `visit_identifier_reference` may cause calling `try_diagnostic_forbid_const_assign` multiple times on the same `SymbolId`.